src/main/groovy/org/transmartproject/batch/highdim/proteomics/platform/FillInUniprotNamesTasklet.groovy

```diff
@@ -0,0 +1,41 @@
+package org.transmartproject.batch.highdim.proteomics.platform
+
+import org.springframework.batch.core.StepContribution
+import org.springframework.batch.core.scope.context.ChunkContext
+import org.springframework.batch.core.step.tasklet.Tasklet
+import org.springframework.batch.repeat.RepeatStatus
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Component
+import org.transmartproject.batch.beans.JobScopeInterfaced
+import org.transmartproject.batch.clinical.db.objects.Tables
+import org.transmartproject.batch.highdim.platform.Platform
+
+/**
+ * Fills in de_protein_annotation.uniprot_name column with data from biomart.bio_marker dictionary (legacy).
+ */
+@Component
+@JobScopeInterfaced
+class FillInUniprotNamesTasklet implements Tasklet {
+
+    @Autowired
+    Platform platform
+
+    @Autowired
+    NamedParameterJdbcTemplate jdbcTemplate
+
+    @Override
+    RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
+        int i = jdbcTemplate.update("""
+                UPDATE ${Tables.PROTEOMICS_ANNOTATION}
+                SET uniprot_name =
+                (SELECT ${Tables.BIO_MARKER}.bio_marker_name
+                 FROM ${Tables.BIO_MARKER}
+                 WHERE ${Tables.BIO_MARKER}.primary_external_id = ${Tables.PROTEOMICS_ANNOTATION}.uniprot_id
+                  AND ${Tables.BIO_MARKER}.bio_marker_type = :marker_type)
+                WHERE gpl_id = :gpl_info
```

Comment:
The best would be to load the 5k something rows into memory and avoid doing the update in the first place.